### DISCUSSION TITLE
Feature idea: tempfiles with meaningful timestamps for difftool

### DISCUSSION THREAD

**Discussion by cforbin1** [gitextensions/gitextensions#10078] [Ideas] [answered]
I have an idea for a mildly-useful feature than may be easy to implement.

TL;DR = set tempfile date/times to author date (or commit date) before handing off to difftool.

DISCUSSION

Today I was comparing some revisions for a code review using an external (3rd party) diff tool, and during the review, suddenly realized I wasn't sure which two revisions I was looking at.  I looked at the filedates (shown prominently by my difftool) but realized those were simply going to be when the temp files were generated.

After I came back to the difftool from looking at Git Extensions for those timestamps, it dawned on me: G.E. knows the author date for both the left hand and right hand revisions when it is writing the temp files for hand-off to the external diff tool.  _Why couldn't it just set the files' creation and modification dates accordlingly?_  That way, you can see those dates from the diff tool.

![image](https://user-images.githubusercontent.com/45664012/180288326-4b18aa90-218e-4504-84d2-116c1096f94f.png)

**Comment by gerhardol** [comment, marked as answer]
Most temporary files are handled by Git, i.e. GE provides the commit sha for the files to Git that creates the tempfiles.
Most difftools do not display the timestamp.
So this would be a quite big change. (Some of the handling is in the MergeTool handling already though. There are a lot of specials there...)

**Reply by cforbin1** [reply]
If this is quite a big change, then I withdraw my request for doing it.
It has value, but not a GREAT DEAL of value, and your time is not infinite, and better spent on high-value features/fixes.

I recommend just dropping this as a "won't implement."
Thanks!!
